app/src/main/java/com/eventyay/organizer/core/organizer/detail/OrganizerDetailPresenter.java

```diff
@@ -34,10 +43,21 @@ public void loadOrganizer(boolean forceReload) {
             .compose(progressiveErroneousRefresh(getView(), forceReload))
             .subscribe(loadedUser -> {
                 this.user = loadedUser;
+                resendVerificationMail
+                    .setEmail(user.email);
                 getView().showResult(user);
             }, Logger::logError);
     }
 
+    public void resendVerificationMail() {
+        compositeDisposable.add(
+            resendVerificationMailModel.resendVerificationMail(resendVerificationMail)
+                .doOnSubscribe(disposable -> getView().showProgress(true))
+                .doFinally(() -> getView().showProgress(false))
+                .subscribe(resendMailResponse -> getView().showSnackbar("Verification Mail Resent"),
+                    throwable -> getView().showError(ErrorUtils.getMessage(throwable).toString())));
```

Comment:
Handle error correctly, don't just show to the user

Reply:
Is it okay now?